src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -218,12 +218,13 @@ def extract_trusted_facts(request_data, facts)
       end
 
       def get_facts_from_pdb(nodename, environment)
+        env = environment || 'production'
```

Comment:
I kind of don't love providing this default in two places (also https://github.com/puppetlabs/puppetserver/blob/master/src/ruby/puppetserver-lib/puppet/server/compiler.rb#L38), but I'm not sure if it's worth passing `environment` separately all the way through...